src/composing/platt_scaling.rs

```diff
@@ -79,23 +88,29 @@ impl<F: Float, O> PlattParams<F, O> {
     /// size in each optimization step. In each attempt the stepsize is halfed until this threshold
     /// is reached. After reaching the threshold the algorithm fails because the desired precision
     /// could not be achieved.
-    pub fn minstep(mut self, minstep: F) -> Self {
+    pub fn minstep(mut self, minstep: F) -> PlattParams<F, O, false> {
         self.minstep = minstep;
 
-        self
+        self.change_check_flag()
     }
 
     /// Set the Hessian's sigma value
     ///
     /// The Hessian matrix is regularized with H' = H + sigma I to avoid numerical issues. This
     /// function set the amount of regularization.
-    pub fn sigma(mut self, sigma: F) -> Self {
+    pub fn sigma(mut self, sigma: F) -> PlattParams<F, O, false> {
         self.sigma = sigma;
 
-        self
+        self.change_check_flag()
     }
 
-    fn validate(&self) -> Result<(), PlattNewtonResult> {
+}
+
+impl<F: Float, O> ParameterCheck for PlattParams<F, O, false> {
+    type Checked = PlattParams<F, O, true>;
+    type Error = PlattNewtonResult;
```

Comment:
Shouldn't this be an error type, not a result type?

Reply:
this is the error enum and should be renamed to `PlattNewtonError`

Reply:
done